sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/ODataDataObject.scala

```diff
@@ -0,0 +1,863 @@
+  }
+
+  def fileExists(path:java.nio.file.Path): Boolean = {
+    Files.exists(path)
+  }
+
+  def fileCreateDirectories(path:java.nio.file.Path) : java.nio.file.Path = {
+    Files.createDirectories(path)
+  }
+
+  def getInstantNow : Instant = {
+    Instant.now()
+  }
+
+  def writeHadoopFile(path: HadoopPath, content: String, filesystem: FileSystem) : Unit = {
+    HdfsUtil.writeHadoopFile(path, content)(filesystem)
+  }
+
+  def sparkDataTypeFromDDL(ddl: String) : DataType = {
+    DataType.fromDDL(ddl)
+  }
+}
+
+
+/**
+ * [[ODataAuthorization]] contains the coordinates and credentials to gain access to the DataSource
+ * @param oauthUrl: URL to the OAuth2 authorization instance like "https://login.microsoftonline.com/{tenant-guid}/oauth2/v2.0/token"
+ * @param clientId: Name of the user (supports secrets providers)
+ * @param clientPassword: Password of the user (supports secret providers)
+ * @param oauthScope: OAuth authorization scope (like https://xxx.crm4.dynamics.com/.default)
+ */
+case class ODataAuthorization(oauthUrl: String
+                              , clientId: StringOrSecret
+                              , clientPassword: StringOrSecret
+                              , oauthScope: String
+                             )
+
+/**
+ * [[ODataResponseBufferSetup]] contains configurations related to the response buffer
+ * @param tempFileBufferType : Name of the buffer type when file buffering is required (can either be "local" or "dbfs")
+ * @param tempFileDirectoryPath : Path to the temporary directory in which the temporary files can be stored
+ * @param memoryToFileSwitchThresholdNumOfChars : Max number of character which can be stored in memory before switching to file buffering. (If set to 0 then no swichting will occur)
+ */
+case class ODataResponseBufferSetup(
+                                     tempFileBufferType: Option[String] = None
+                                     , tempFileDirectoryPath: Option[String] = None
+                                     , memoryToFileSwitchThresholdNumOfChars: Option[Long] = None
+                                   ) {
+
+  /* Late-Arriving property
+   */
+  private var tableName: String = ""
+  def getTableName: String = this.tableName
+  def setTableName(tableName: String): Unit = this.tableName = tableName
+
+}
+
+/**
+ * [[ODataBearerToken]] contains the current bearer token and a method to check whether the token is still valid.
+ * @param token : The token string
+ * @param expiresAt : Instant at which the token expires
+ */
+case class ODataBearerToken(token: String, expiresAt:java.time.Instant) {
+
+  def isExpired : Boolean = {
+    expiresAt.isBefore(Instant.now())
+  }
+}
+
+/**
+ * [[ODataResponseBuffer]] is the base class for all other response buffer implementations
+ * @param setup: The setup object which contains the configuration for this response buffer
+ */
+abstract class ODataResponseBuffer(setup: ODataResponseBufferSetup, context: ActionPipelineContext) {
```

Comment:
I suggest to move the buffer interface and implementation to a separate file in package  io.smartdatalake.util.... 
And could we make this buffer Logic OData agnostic, so it can be reused for other DataObjects?